dialects/linux/dproc.c

```diff
@@ -1395,6 +1397,50 @@ process_id(idp, idpl, cmd, uid, pid, ppid, pgid, tid, tcmd)
 	return(0);
 }
 
+/* compare namespace of this lsof proces and the target process */
+
+int
+compare_namespace(ns)
+        char *ns;                /* namespace */
+{
+        char nspath_self[NS_PATH_LENGTH], nspath_target[NS_PATH_LENGTH];
+        char ns_self[NS_LENGTH], ns_target[NS_LENGTH];
+        ssize_t n_self, n_target;
+
+        sprintf(nspath_self, "/proc/self/ns/%s", ns);
+        n_self = readlink(nspath_self, ns_self, NS_LENGTH);
+        if (n_self < 0)
+            return -1;
+
+        sprintf(nspath_target, "/proc/%d/ns/%s", Lp->pid, ns);
```

Comment:
Similar buffer overflow issue here.

Reply:
I also thought sprintf with no limit could be an issue.
However, I could not find a threat situation.

I have just found one:

```
$ mkdir x
$ mkdir x/lib64
$ sudo mount -o bind /lib64 x/lib64
$ mkdir x/lib                      
$ sudo mount -o bind /lib x/lib    
$ mkdir x/bin                  
$ sudo mount -o bind /bin x/bin
$ sudo chroot x /bin/bash
bash-5.1# mkdir /proc
bash-5.1# mkdir /proc/0123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789/
bash-5.1# mkdir /proc/0123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789/ns
bash-5.1# touch /proc/0123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789/ns/mnt
bash-5.1# ls -l /proc/0123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789/ns/mnt
-rw-r--r--. 1 0 0 0 Jul 23 04:40 /proc/0123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789/ns/mnt
```

With overlayfs, we can make a more realistic example.
Using sprintf is better.

Reply:
Considered on this situation, it's necessary and practical to handle this with snprintf. Thanks a lot.